### DISCUSSION TITLE
How do I apply filters when creating a new item?

### DISCUSSION THREAD

**Discussion by jhh78** [pocketbase/pocketbase#4291] [Q&A] [answered]
Greetings.
I am trying to create an item for the collection.
When creating an item in a collection, I want to prevent duplicate items from being entered.

`nickname != @request.data.nickname`
I used this, but it doesn't work properly.

Is this possible with the filter functions provided?
If there is a better way, please let me know.

**Comment by ganigeorgiev** [comment, marked as answer]
You don't need an API rule for this.

For unique field validations you can create a unique index for your field from the _Admin UI > Collections > Edit collection panel > "Unique constraints and indexes" (under the fields)_:

![unique_index](https://i.imgur.com/SURee8H.png)
